migration_guides/v0.10.2_to_v1.0.0_nearcore.md

```diff
@@ -0,0 +1,375 @@
+index 30af36fb9..d5a6632ed 100644
+--- a/core/primitives/src/receipt.rs
++++ b/core/primitives/src/receipt.rs
+@@ -1,11 +1,11 @@
+-use crate::borsh::maybestd::collections::HashMap;
++use std::collections::HashMap;
+```
+
+Otherwise, we would've imported from `hashbrown`:
+
+```diff
+-use crate::borsh::maybestd::collections::HashMap;
++#[cfg(feature = "std")]
++use std::collections::HashMap;
++#[cfg(not(feature = "std"))]
++use hashbrown::HashMap;
+```
+
+## 3. next we encounter a bunch of similar errors in `near-primitives` with `#[borsh_init(...)]`:
+
+```bash
+ 1  error: cannot find attribute `borsh_init` in this scope                                 ▐
+    --> core/primitives/src/block_header.rs:267:3                                           ▐
+     |                                                                                      ▐
+ 267 | #[borsh_init(init)]                                                                  ▐
+     |   ^^^^^^^^^^ help: a derive helper attribute with a similar name exists: `borsh_skip`▐
+```
+
+The syntax of this attribute has changed. We change all of these occurencies according to
+`#[borsh(init=<initialization_method>)]` syntax. The following diff is shortened to first and last
+occurencies:
+
+```diff
+diff --git a/core/primitives/src/block_header.rs b/core/primitives/src/block_header.rs
+index 38491b52c..84ab48238 100644
+--- a/core/primitives/src/block_header.rs
++++ b/core/primitives/src/block_header.rs
+@@ -266,3 +266,3 @@ impl ApprovalMessage {
+ #[derive(BorshSerialize, BorshDeserialize, serde::Serialize, Debug, Clone, Eq, PartialEq)]
+-#[borsh_init(init)]
++#[borsh(init=init)]
+ pub struct BlockHeaderV1 {
+...
+diff --git a/core/primitives/src/transaction.rs b/core/primitives/src/transaction.rs
+index 912120b56..2de7a1d52 100644
+--- a/core/primitives/src/transaction.rs
++++ b/core/primitives/src/transaction.rs
+@@ -58,3 +58,3 @@ impl Transaction {
+ )]
+-#[borsh_init(init)]
++#[borsh(init=init)]
+ pub struct SignedTransaction {
+```
+
+## 4. next we encounter 2 errors in `near-primitives` package similar to those in point 2.:
+
+```bash
+ 1  error[E0433]: failed to resolve: could not find `maybestd` in `borsh`
+    --> core/primitives/src/action/delegate.rs:119:41
+     |
+ 119 |         fn deserialize_reader<R: borsh::maybestd::io::Read>(
+     |                                         ^^^^^^^^ could not find `maybestd` in `borsh`
+                                                                                             
+ 2  error[E0433]: failed to resolve: could not find `maybestd` in `borsh`
+    --> core/primitives/src/action/delegate.rs:121:50
+     |
+ 121 |         ) -> ::core::result::Result<Self, borsh::maybestd::io::Error> {
+     |                                                  ^^^^^^^^ could not find `maybestd` i
+ n `borsh`
+```
+
+As `near-primitives` crate is not supposed to be used in `no_std` context, we can
+replace import with `std::io`:
```

Comment:
Let's focus only on new classes of "errors" and skip the ones that were covered earlier.

Reply:
It's the same class as one of above, but i specifically mentioned it with a ready recipe, as there's not much else, besides `hashbrown` and `nostd_io` in `borsh::__private` that branches on `std` support.
People also have been using `borsh::maybestd` for things, that are only available in `std`, like `std::sync::Arc`.

```diff
+#[cfg(feature = "std")]
+use std::io;
+#[cfg(not(feature = "std"))]
+use borsh::nostd_io as io;
```